package.json

```diff
@@ -569,7 +569,7 @@
   },
   "scripts": {
     "vscode:prepublish": "npm run esbuild-base -- --minify && npm run esbuild-server -- --minify",
-    "esbuild-base": "esbuild ./src/extension.ts --bundle --outfile=out/extension.js --external:vscode --format=cjs --platform=node",
+    "esbuild-base": "esbuild ./src/extension.ts --bundle --outfile=out/extension.js --external:vscode --external:jsonc-parser --format=cjs --platform=node",
```

Comment:
It gets installed into the node_modules the external allows the path stuff to resolved properly

Reply:
Original issue without the --external:

Activating extension 'vectorgroup.vectorcasttestexplorer' failed: Cannot find module './impl/format' Require stack: - c:\RDS\vector-vscode-vcast\out\extension.js - c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js - c:\Program Files\Microsoft VS Code\resources\app\out\bootstrap-amd.js - c:\Program Files\Microsoft VS Code\resources\app\out\bootstrap-fork.js - .

--external is the wrong solution

Reply:
This happens because of how microsoft write their packages which unfortunately does not make them easily compatible with bundlers like esbuild. When writing CommonJS packages they tend to wrap things in such a way that the original require function is called indirectly and not directly which means in the source the bundler is not aware it is a real call to require. As such, we would need to provide the bundler with more context to get around this. Thankfully, the ESM module that is provided is written as you'd expect and we can tell esbuild to prefer using the ESM module by specifying `--main-fields=module,main` after the `esbuild` command. This is only an issue with `extension.js` building so I recommend changing
```
"esbuild-base": "esbuild ./src/extension.ts --bundle --outfile=out/extension.js --external:vscode --format=cjs --platform=node",
```
to
```
"esbuild-base": "esbuild ./src/extension.ts --bundle --outfile=out/extension.js --external:vscode --format=cjs --platform=node --main-fields=module,main",
```
This is not a perfect solution and it has potential to break things in our dependencies do not conform properly to supporting ESM according to sources, but I have tested it here and it seems to work.

References:
- https://stackoverflow.com/questions/69208729/how-to-get-esbuild-to-include-internal-dependencies-when-using-typescript
- https://github.com/evanw/esbuild/issues/1619

Reply:
Thanks @ojcole!